src/GoogleFitOnFhir/Clients/GoogleFit/Handlers/GoogleFitHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using EnsureThat;
+using GoogleFitOnFhir.Clients.GoogleFit.Responses;
+using GoogleFitOnFhir.Common;
+using GoogleFitOnFhir.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Common.Handler;
+
+namespace GoogleFitOnFhir.Clients.GoogleFit.Handlers
+{
+    public class GoogleFitHandler : IResponsibilityHandler<RoutingRequest, Task<IActionResult>>
+    {
+        private const string GoogleFitAuthorizeRequest = "api/googlefit/authorize";
+        private const string GoogleFitCallbackRequest = "api/googlefit/callback";
+        private IAuthService _authService;
+        private IUsersService _usersService;
+        private readonly ILogger<GoogleFitHandler> _logger;
+
+        public GoogleFitHandler(IAuthService authService, IUsersService usersService, ILogger<GoogleFitHandler> logger)
+        {
+            _authService = authService;
+            _usersService = usersService;
+            _logger = logger;
+        }
+
+        public Task<IActionResult> Evaluate(RoutingRequest request)
+        {
+            var path = EnsureArg.IsNotNullOrWhiteSpace(request.HttpRequest.Path.Value?[1..]);
+
+            if (path.StartsWith(GoogleFitAuthorizeRequest))
+            {
+                return Authorize(request);
+            }
+            else if (path.StartsWith(GoogleFitCallbackRequest))
+            {
+                return Callback(request);
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private async Task<IActionResult> Authorize(RoutingRequest request)
+        {
+            var token = EnsureArg.IsNotDefault(request.Token);
```

Comment:
We don't normally check for cancellation token default values. This is because cancellation tokens in many cases are optional, and in tests, many times we just pass CancellationToken.None.